src/OpenFeature/SharedHookContext.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+
+namespace OpenFeature
+{
+    internal class SharedHookContext<T>(
+        string? flagKey,
+        T defaultValue,
+        FlagValueType flagValueType,
+        ClientMetadata? clientMetadata,
+        Metadata? providerMetadata)
+    {
+        public string FlagKey { get; } = flagKey ?? throw new ArgumentNullException(nameof(flagKey));
+
+        public T DefaultValue { get; } = defaultValue;
+
+        public FlagValueType FlagValueType { get; } = flagValueType;
+
+        public ClientMetadata ClientMetadata { get; } =
+            clientMetadata ?? throw new ArgumentNullException(nameof(clientMetadata));
+
+        public Metadata ProviderMetadata { get; } =
+            providerMetadata ?? throw new ArgumentNullException(nameof(providerMetadata));
+
+        public HookContext<T> ToHookContext(EvaluationContext? evaluationContext)
```

Comment:
Convenient for using the shared context as a template and instantiating the per-instance data.